lib/db.mysql.js

```diff
@@ -31,36 +55,17 @@ if (api.mysql) {
         setTimeout(() => {
           if (database.retryCounter <= database.retryCount) {
             api.db.mysql.open(database, callback);
+          } else {
+            callback(new Error('Cannot open connection'));
           }
         }, database.retryInterval);
+        return;
       }
       database.retryCounter = 0;
+      setupConnection(database, connection, callback);
```

Comment:
You still pass `callback` which will be called on connection error
https://github.com/metarhia/impress/blob/1a6197c5b9468d2782fcc80f9eaf39c572311cbf/lib/db.mysql.js#L19-L26
in addition to being called at the end of this function
https://github.com/metarhia/impress/blob/1a6197c5b9468d2782fcc80f9eaf39c572311cbf/lib/db.mysql.js#L67

Reply:
Yes, that will lead to the re-establishment of a database connection after error, that's the current behavior, though all other DB connectors do not reconnect in case of error.
I can remove this if there is a consensus on this topic. Personally, I think it should align with other DB connectors (remove api.db.mysql.open from 'error' in mysql)